### PR TITLE
Refactor get/set members to SynMemberDefn.GetSetMember.

### LINKED ISSUE #13273: Better representation of get/set member
Giving the following code and AST:
```fsharp
type Foo() =
    member this.MyReadWriteProperty
        with get () =   //comment get 
            myInternalValue
        and set (value) =   // comment set
            myInternalValue <- value
```

```fsharp
SynTypeDefn
   (SynComponentInfo
      ([], None, [], [Foo],
       PreXmlDoc ((2,0), FSharp.Compiler.Xml.XmlDocCollector),
       false, None, tmp.fsx (2,5--2,8)),
    ObjectModel
      (Unspecified,
       [ImplicitCtor
          (None, [], SimplePats ([], tmp.fsx (2,8--2,10)), None,
           PreXmlDoc ((2,8), FSharp.Compiler.Xml.XmlDocCollector),
           tmp.fsx (2,5--2,8));
        Member
          (SynBinding ...
        Member
          (SynBinding ...
```

It is very confusing that there are two `SynMemberDefn.Member` created for this in the untyped tree.

**Describe the solution you'd like**

Could we have a special case in `SynMemberDefn` that represents this better?
The name `SynMemberDefn.ReadWriteProperty`comes to mind, but I'm open to suggestions.

**Describe alternatives you've considered**

Both `Member`s have the same range, so you can hack around it.

**Additional context**

related: #11481

//cc @dsyme

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #13273.

### CODE REVIEW

**Comment by nojaf** [comment]
@dsyme ready for review!

File: src/Compiler/Service/ServiceParseTreeWalk.fs [hunk 1]
```diff
@@ -824,47 +824,19 @@ module SyntaxTraversal =
...
-                | _ ->
-#if DEBUG
-                    assert false // more than 2 members claim to have the same range, this indicates a bug in the AST
-                    failwith "bug in AST"
-#else
-                    // swallow AST error and recover silently
-                    None
-#endif
-            )
+            |> Seq.map (fun mb ->
```
Comment (dsyme) [inline src/Compiler/Service/ServiceParseTreeWalk.fs:827]: Great to see this code get simpler!

File: src/Compiler/Service/ServiceStructure.fs [hunk 2]
```diff
@@ -587,6 +587,15 @@ module Structure =
                 parseAttributes attrs
                 parseBinding binding
 
+            | SynMemberDefn.GetSetMember (getBinding, setBinding, m, _) ->
+                (Option.map (fun b -> SynMemberDefn.Member(b, m))
+                 >> Option.iter (parseSynMemberDefn objectModelRange))
```
Comment (dsyme) [inline src/Compiler/Service/ServiceStructure.fs:592]: I'll change this to use pipelining

**Review by dsyme — APPROVED** [review]

File: tests/FSharp.Compiler.ComponentTests/Conformance/ClassTypes/GetSetMembers/GetSetMembers.fs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace FSharp.Compiler.ComponentTests.Conformance.ClassTypes
+
+open Xunit
+open FSharp.Test.Compiler
+
+module GetSetMembers =
+
+    [<Fact>]
+    let WithGetAndSet() =
```
[the remark this answers is not in the document]
Comment (nojaf) [reply tests/FSharp.Compiler.ComponentTests/Conformance/ClassTypes/GetSetMembers/GetSetMembers.fs:11]: I removed these tests in https://github.com/dotnet/fsharp/pull/13435.

**Pull request merged**